transcom/mymove
# INT-B-20554 Add Boat Shipments to Prime Sim

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20554)

## Summary

This branch adds boat shipments to the Prime Simulator. Prime should now be able to add a boat shipment to a move by using the "Create Shipment" button, then choosing either "BOAT_HAULAWAY" or "BOAT_TOWAWAY", they will then be given a form to fill out that includes boat-specific info, such as make, model, dimensions, trailer etc., as well as all of the usual HHG shipment information (Diversion, Addresses, Weight, etc.).

It also adds feature flag checks so that these will only be seen by the Prime if the boat flag is enabled. There are also checks on the backend for this flag to ensure boat shipments sent via the API will not be processed if the flag is not enabled.

Finally, it adds frontend fields for the shipment info card component to display boat-specific info to the Prime if the shipment being viewed is a boat shipment.

### How to test

1. Login as a Prime user
2. Access any shipment that is available
3. Click "Create Shipment" in the shipment details section.
4. Ensure that you are able to create a shipment with both haul-away and tow-away types.
5. You can grab the shipment ID and check in the database that the fields are actually populated, or you can check the shipment card back at the move page, you should see the boat specific fields as well as the HHG generic fields show up.
6. After going through the usual process (approving as SC, using TIO/TOO to approve boat shipments), you should see the prime card have all of the fields filled out. Some fields will be blank before this process since they rely on that information (i.e. actual pickup date, actual weight, etc.)

Screenshots of cards in SC/TIO/TOO "Move details" page, of a shipment created solely via Prime Sim:

![boat_card_1](https://github.com/user-attachments/assets/d9cd4b44-c361-4e20-949f-6f6b3157daae)
![boat_card_2](https://github.com/user-attachments/assets/32e502ec-327b-461c-b8c1-d20717f1cbc4)
![boat_card_3](https://github.com/user-attachments/assets/c7b47897-492b-4d9f-9bef-0ed3696c303e)

Example of boat details in the Prime Sim:
![prime_boat_card_after_process](https://github.com/user-attachments/assets/21f6c5c7-65c6-47ea-8aba-d48734c6497e)
